ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactHorizontalScrollView.java

```diff
@@ -1239,6 +1288,97 @@ private void updateStateOnScroll() {
     updateStateOnScroll(getScrollX(), getScrollY());
   }
 
+  /**
+   * Called when a mContentView's layout has changed. Fixes the scroll position depending on
+   * maintainVisibleContentPosition
+   */
+  @Override
+  public void onLayoutChange(
+      View v,
+      int left,
+      int top,
+      int right,
+      int bottom,
+      int oldLeft,
+      int oldTop,
+      int oldRight,
+      int oldBottom) {
+    if (mContentView == null) {
+      return;
+    }
+
+    if (this.mMaintainVisibleContentPositionData != null) {
+      scrollMaintainVisibleContentPosition();
+    }
+  }
+
+  /**
+   * Called when maintainVisibleContentPosition is used and after a scroll. Finds the first
+   * completely visible view in the ScrollView and stores it for later use.
+   */
+  private void computeFirstVisibleItemForMaintainVisibleContentPosition() {
+    ReactScrollViewMaintainVisibleContentPositionData maintainVisibleContentPositionData =
+        mMaintainVisibleContentPositionData;
+    if (maintainVisibleContentPositionData == null) return;
+
+    int currentScrollX = getScrollX();
+    int minIdx = maintainVisibleContentPositionData.minIndexForVisible;
+
+    ReactViewGroup contentView = (ReactViewGroup) getChildAt(0);
+    if (contentView == null) return;
+
+    for (int i = minIdx; i < contentView.getChildCount(); i++) {
```

Comment:
It would be better to use binary search here to reduce computational complexity.

Normally this is not a big deal, but here we are calling this method on every scroll ends (with the debounce it's better, but still will call this a lot). If we have many items in the list, this could lead to a slow perf issue.